PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchViewModel.swift

```diff
@@ -110,32 +112,38 @@ class SearchViewModel: ObservableObject {
     func clear() {
         searchResults = []
         subscriptions = []
+        savesLocalSearch.clear()
+        savesOnlineSearch.clear()
+        archiveOnlineSearch.clear()
+        allOnlineSearch.clear()
     }
 
-    func clearCache() {
-        cachedSearchResults = [
-            .saves: [:],
-            .archive: [:],
-            .all: [:]
-        ]
-    }
-
-    private func retrieveCachedResults() -> Bool {
-        if let cachedItems = cachedSearchResults[selectedScope]?[currentSearchTerm], !cachedItems.isEmpty {
-            self.searchResults = cachedItems
-            return true
-        }
-        return false
-    }
-
-    private func submitOnlineSearch(with term: String) {
-        clear()
-        searchService.results.receive(on: DispatchQueue.main).sink { [weak self] items in
-            guard let self else { return }
-            self.searchResults = items.compactMap { SearchItem(item: $0) }
-        }.store(in: &subscriptions)
-        Task {
-            await searchService.search(for: term, scope: selectedScope)
+    private func submitSearch(with term: String) {
```

Comment:
Recommend making state a function parameter, rather than checking an instance variable. By removing a local variable, you can simplify your control logic, decrease the likelihood of errors, and make the function easier to debug and test. You can expand on that logic to make this a pure function if you want. One of the key advantages being that testing pure functions is trivial.

Reply:
Not too sure if I understand this fully. I updated the PR with new changes, but let me know if I missed something and happy to get on a call. For a pure function, it seems that the same input always return the same result, but we don't return the results due the nature of pagination from the online search.

Reply:
You are correct. 

This looks like a meaningful improvement to me.